kyo-prelude/shared/src/main/scala/kyo/kernel/Context.scala

```diff
@@ -1,32 +1,48 @@
 package kyo.kernel
 
+import Context.internal.*
 import kyo.Tag
 import kyo.bug
 
-opaque type Context = Map[Tag[Any], AnyRef]
+opaque type Context = Map[Tag[Any] | IsolationFlag, AnyRef]
```

Comment:
The solution uses an `IsolationFlag` placeholder entry in the map indicating if the context has isolated effects and needs to `filter` in `inherit`